app/src/main/java/org/mozilla/focus/web/ClassicWebViewProvider.java

```diff
@@ -124,38 +127,40 @@ private static void configureDefaultSettings(Context context, WebSettings settin
         settings.setSavePassword(false);
     }
 
-    public static void applyAppSettings(Context context, WebSettings settings, WebView webView) {
+    @Override
+    public void applyAppSettings(@NotNull Context context, @NotNull WebSettings webSettings, @NotNull SystemWebView systemWebView) {
         // We could consider calling setLoadsImagesAutomatically() here too (This will block images not loaded over the network too)
-        settings.setBlockNetworkImage(Settings.getInstance(context).shouldBlockImages());
-        settings.setJavaScriptEnabled(!Settings.getInstance(context).shouldBlockJavaScript());
-        CookieManager.getInstance().setAcceptThirdPartyCookies(webView, !Settings.getInstance
+        webSettings.setBlockNetworkImage(Settings.getInstance(context).shouldBlockImages());
+        webSettings.setJavaScriptEnabled(!Settings.getInstance(context).shouldBlockJavaScript());
+        CookieManager.getInstance().setAcceptThirdPartyCookies(systemWebView, !Settings.getInstance
                 (context).shouldBlockThirdPartyCookies());
         CookieManager.getInstance().setAcceptCookie(!Settings.getInstance(context)
                 .shouldBlockCookies());
     }
 
+    @Override
     @SuppressLint("SetJavaScriptEnabled") // We explicitly want to enable JavaScript
-    public static void disableBlocking(WebSettings settings, WebView webView) {
-        settings.setBlockNetworkImage(false);
-        settings.setJavaScriptEnabled(true);
-        CookieManager.getInstance().setAcceptThirdPartyCookies(webView, true);
+    public void disableBlocking(@NotNull WebSettings webSettings, @NotNull SystemWebView systemWebView) {
+        webSettings.setBlockNetworkImage(false);
+        webSettings.setJavaScriptEnabled(true);
+        CookieManager.getInstance().setAcceptThirdPartyCookies(systemWebView, true);
         CookieManager.getInstance().setAcceptCookie(true);
     }
 
-    public static void requestDesktopSite(WebSettings settings) {
+    public void requestDesktopSite(WebSettings settings) {
         settings.setUserAgentString(toggleDesktopUA(settings, true));
         settings.setUseWideViewPort(true);
     }
 
-    public static void requestMobileSite(Context context, WebSettings settings) {
+    public void requestMobileSite(Context context, WebSettings settings) {
         settings.setUserAgentString(toggleDesktopUA(settings, false));
         settings.setUseWideViewPort(false);
     }
 
     /**
      * Build the browser specific portion of the UA String, based on the webview's existing UA String.
      */
-    @VisibleForTesting static String getUABrowserString(final String existingUAString, final String focusToken) {
+    public String getUABrowserString(final String existingUAString, final String focusToken) {
```

Comment:
\[PMD\] INFO: The method 'getUABrowserString\(String, String\)' has a cyclomatic complexity of 5\.